[chore]  update core modules

mx-psi opened the pull request:
Signed-off-by: Pablo Baeyens <pablo.baeyens@datadoghq.com>**Description:** <Describe what has changed.>
<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->

Updates core modules to open-telemetry/opentelemetry-collector@e7ce1d50fb5e. To do this I had to tidy manually by doing:

```
make for-all CMD="go mod edit -require=go.opentelemetry.io/collector/internal/featuregates@v0.0.0-20240705161705-b127da089038 "
make gotidy
```

to work around #33954

**Link to tracking Issue:** Fixes #33954

mx-psi committed:
[chore] multimod update stable modules

Signed-off-by: Pablo Baeyens <pablo.baeyens@datadoghq.com>

mx-psi committed:
[chore] multimod update beta modules

Signed-off-by: Pablo Baeyens <pablo.baeyens@datadoghq.com>

mx-psi committed:
[chore] Explicitly require internal/featuregates and tidy

I ran the following commands for this commit

```
make for-all CMD="go mod edit -require=go.opentelemetry.io/collector/internal/featuregates@v0.0.0-20240705161705-b127da089038 "
make gotidy
```

mx-psi committed:
make genotelcontribcol && make genoteltestbedcol

codeboten reviewed (commented):
Thanks @mx-psi, looks like a make genotelcontribcol is needed. Is this a problem anyone pulling in the collector as a dependency will run into as well w/ the new release?

codeboten commented:
> Is this a problem anyone pulling in the collector as a dependency will run into as well w/ the new release?

If so, it might be worth renaming the internal module to avoid the issue altogether

mx-psi committed:
Merge remote-tracking branch 'origin/main' into mx-psi/update-otel-configtls

mx-psi committed:
Fix bad merge

mx-psi commented:
Filed #10569 for discussing the rename but I think we should still merge this to unblock core updates

mx-psi committed:
Merge remote-tracking branch 'origin/main' into mx-psi/update-otel-configtls

mx-psi committed:
make generate

This is needed because of open-telemetry/opentelemetry-collector@d545fb6a102caf06d3b15d2dba5e652bcbb000d4

mx-psi committed:
Merge branch 'main' into mx-psi/update-otel-configtls
